tests/fips/openssl/dirmngr_setup.pm

```diff
@@ -46,7 +46,8 @@ sub dirmngr_setup {
     assert_script_run "curl --silent " . data_url('openssl/root-ca/root-ca.conf') . " --output $ca_cfg";
 
     # Create root ca certificate
-    assert_script_run("openssl req -new -config $ca_cfg -out $ca_csr -keyout $ca_key -passout 'pass:$ssl_pwd'");
+    assert_script_run("openssl genrsa -aes256 -out $ca_key -passout 'pass:$ssl_pwd' 2048");
```

Comment:
The additional command might not be required. According to `man openssl-req`:
```
-newkey arg
           This option is used to generate a new private key unless -key is given.  It is subsequently used as if it was given using the -key option.

           This option implies the -new flag to create a new certificate request or a new certificate in case -x509 is given.

           The argument takes one of several forms.

           [rsa:]nbits generates an RSA key nbits in size.  If nbits is omitted, i.e., -newkey rsa is specified, the default key size specified in the configuration file with the default_bits option is used if present, else 2048.

           All other algorithms support the -newkey algname:file form, where file is an algorithm parameter file, created with "openssl genpkey -genparam" or an X.509 certificate for a key with appropriate algorithm.

           param:file generates a key using the parameter file or certificate file, the algorithm is determined by the parameters.

           algname[:file]  generates a key using the given algorithm algname.  If a parameter file file is given then the parameters specified there are used, where the algorithm parameters must match algname.  If algorithm parameters are not given, any necessary parameters should be specified via the -pkeyopt
           option.

           dsa:filename generates a DSA key using the parameters in the file filename. ec:filename generates EC key (usable both with ECDSA or ECDH algorithms), gost2001:filename generates GOST R 34.10-2001 key (requires gost engine configured in the  configuration  file).  If  just  gost2001  is  specified  a
           parameter set should be specified by -pkeyopt paramset:X
```

Edit: So `-newkey` instead of `-new` should allow to specify `rsa:2048`.

Reply:
OK, maybe I made a mistake here and there is no option to specify AES when using req...

Reply:
There is another benefit of not doing everything at once. It took me a while to understand what exactly was failing in the request creation call.